pkg/controller/worker/machines_test.go

```diff
@@ -640,6 +641,107 @@ var _ = Describe("Machines", func() {
 					Expect(err).NotTo(HaveOccurred())
 					Expect(result).To(Equal(machineDeployments))
 				})
+
+				It("should delete the all old AWSMachineClasses", func() {
+					// mockAWSMachineClassList is the test object used to track the
+					// list of AWSMachineClass objects
+					awsMachineClassList := machinev1alpha1.AWSMachineClassList{
+						Items: []machinev1alpha1.AWSMachineClass{
+							{
+								ObjectMeta: metav1.ObjectMeta{
+									Name: "test-aws-machine-class-0",
+								},
+							},
+							{
+								ObjectMeta: metav1.ObjectMeta{
+									Name: "test-aws-machine-class-1",
+								},
+							},
+						},
+					}
+
+					// Initialize mocking functions
+					expectGetSecretCallToWork(c, awsAccessKeyID, awsSecretAccessKey)
+					expectDeleteAllCallToWork(c, &awsMachineClassList)
+
+					workerDelegate, _ := NewWorkerDelegate(common.NewClientContext(c, scheme, decoder), chartApplier, "", w, cluster)
+					chartApplier.
+						EXPECT().
+						Apply(
+							context.TODO(),
+							filepath.Join(aws.InternalChartsPath, "machineclass"),
+							namespace,
+							"machineclass",
+							kubernetes.Values(machineClasses),
+						).
+						Return(nil)
+
+					err := workerDelegate.DeployMachineClasses(context.TODO())
+					Expect(err).NotTo(HaveOccurred())
+
+					// Should have cleaned up all the old AWSMachineClasses
+					// If the actual code was called the wrong object kind,
+					// the desired object deletion wouldn't have occured.
+					Expect(len(awsMachineClassList.Items)).To(Equal(0))
+				})
+
+				It("should delete all the AWSMachineClasses as parameter is AWSMachineClass", func() {
+					// awsMachineClassList is the test object used to track the
+					// list of AWSMachineClass objects
+					testNamespace := "test-ns"
+					awsMachineClassList := machinev1alpha1.AWSMachineClassList{
```

Comment:
This code fragment is duplicated across the 3 tests. You could extract to a var and reuse it.

Reply:
Done.